packages/orchestration/src/utils/chainHub.js

```diff
@@ -0,0 +1,86 @@
+import { makeScalarBigMapStore } from '@agoric/vat-data';
+import { E } from '@endo/far';
+import { M } from '@endo/patterns';
+import { CosmosChainInfoShape, IBCConnectionInfoShape } from '../typeGuards.js';
+
+/**
+ * @import {NameHub} from '@agoric/vats';
+ * @import {CosmosChainInfo, IBCConnectionInfo} from '../cosmos-api.js';
+ * @import {Remote} from '@agoric/internal';
+ */
+
+// TODO define key hierarchy in shared constants
+/** agoricNames key for ChainInfo hub */
+export const CHAIN_KEY = 'chain';
+/** namehub for connection info */
+export const CONNECTIONS_KEY = 'connections';
+
+/** @param {Remote<NameHub>} agoricNames */
+export const makeChainHub = agoricNames => {
+  // XXX not reachably-durable? really?? take a zone arg?
```

Comment:
Seems reasonable and cheap to allow the caller to choose
```suggestion
export const makeChainHub = (agoricNames, zone = makeHeapZone()) => {
```

we can iterate on what the best way to use it is. it might differ from contract to contract.